### ISSUE TITLE
segfault on Find backwards with empty playlist

### ISSUE THREAD

**Issue by rtoijala** [arybczak/ncmpcpp#142]
Steps to reroduce:

1. Run ncmpcpp
2. Press '1' to switch to playlist view
3. Press 'c' to clear the playlist
4. Press '+' for Find backwards
5. Enter any characacter, e.g. 'a'
6. Press enter

ncmpcpp segfaults.

Archlinux x86_64
```
$ ncmpcpp --version
ncmpcpp 0.7.4

optional screens compiled-in:
 - tag editor
 - tiny tag editor
 - artist info
 - outputs
 - visualizer
 - clock

encoding detection: enabled
built with support for: curl fftw ncurses taglib unicode

```

**Comment by Tmplt** [comment]
I can't reproduce this on `0.8_dev`
```
ncmpcpp 0.8_dev

optional screens compiled-in:
 - tag editor
 - tiny tag editor
 - artist info

encoding detection: enabled
built with support for: curl ncurses taglib unicode
```

**Comment by arybczak** [comment]
See https://bugs.musicpd.org/view.php?id=4532, it's most likely the same underlying issue.

**Comment by arybczak** [comment]
Or if it's not, please provide a backtrace.

**Comment by rtoijala** [comment]
I'm not sure if it's the same issue, so here's the backtrace: 
[bt.txt](https://github.com/arybczak/ncmpcpp/files/390073/bt.txt)

This was created with
```
ncmpcpp 0.8_dev

optional screens compiled-in:
 - tag editor
 - tiny tag editor
 - artist info
 - outputs
 - visualizer
 - clock

encoding detection: enabled
built with support for: curl fftw ncurses taglib unicode
```
built with `-g -O0` from git dcbb3ab7effaa558a4022f5daf6f4ee6d4339f17
on Arch Linux x64 with icu 57.1 and boost 1.60.0

**Comment by arybczak** [comment]
Fixed by e657089b30742f395f4e07db73e2ddd24384c5cc.
